packages/yourloops/test/components/dialogs/pdf-print-options.test.tsx

```diff
@@ -0,0 +1,219 @@
+ *
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import React from "react";
+import ReactDOM from "react-dom";
+import { act } from "react-dom/test-utils";
+import * as sinon from "sinon";
+import { expect } from "chai";
+
+import DialogPDFOptions, { Presets, PrintPDFOptions } from "../../../components/dialogs/pdf-print-options";
+
+function testDialogPDFOptions() {
+  const MIN_DATE = "2020-01-01";
+  const MAX_DATE = "2022-12-31";
+  const handleResult = sinon.stub<[PrintPDFOptions], void>();
+  let container: HTMLDivElement | null = null;
+
+  beforeEach(() => {
+    container = document.createElement("div");
+    document.body.appendChild(container);
+    handleResult.resetHistory();
+  });
+  afterEach(() => {
+    if (container) {
+      ReactDOM.unmountComponentAtNode(container);
+      document.body.removeChild(container);
+      container = null;
+    }
+  });
+
+  const render = (open = true, minDate = MIN_DATE, maxDate = MAX_DATE) => {
+    return act(() => {
+      return new Promise((resolve) => {
+        ReactDOM.render(
+          <DialogPDFOptions open={open} minDate={minDate} maxDate={maxDate} onResult={handleResult} />, container, resolve);
+      });
+    });
+  };
+
+  const expectSelectedPreset = (preset: Presets | null) => {
+    const button1w = document.getElementById("pdf-options-button-one-week");
+    const button2w = document.getElementById("pdf-options-button-two-weeks");
+    const button4w = document.getElementById("pdf-options-button-four-weeks");
+    const button3m = document.getElementById("pdf-options-button-three-months");
+
+    const button1wSelected = button1w.getAttribute("aria-selected") === "true";
```

Comment:
Won't work, it's not the same.
!!"false" = true
!!"true" = true

Reply:
I think it will work, if the attribute not exists, the method returns either null of an empty string. See the [doc here](If the given attribute does not exist, the value returned will either be null or "" (the empty string);). 
const string = ""
!!string returns false

Reply:
And for the naming, it's you ask for a very long name, I'm too lazy to change them